### PR TITLE
Fix loading of questions which user has collection but not data permissions

### LINKED ISSUE #4200: Users can't see question which they have collection permission but not data permission
As reported in this [Discourse thread](http://discourse.metabase.com/t/collection-permissions-cant-view-questions-to-download-data-or-view-line-graphs/1051), if a user group has **no access** to a database or table, but **does** have access to a collection which contains saved questions based on that data, they currently are correctly seeing those questions listed in the collection(s) for which they have access, and are also correctly seeing those cards on dashboards.

However, they are **not** currently able to view those questions' detail views. The specced behavior is that collection permissions should supersede data access permissions. Therefore, if a user group has View permission for a collection, they should be able to view the question detail page, _even if they don't have data access permission_ for the underlying data. In this scenario, they should also probably not be shown the query builder bar, or perhaps should see a disabled version of the query builder.

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #4200 

Also changes how we show 403 errors, so we don't redirect / change the URL (which tends to break the back button)

### CODE REVIEW

**Comment by tlrobinson** [comment]
@metabase/core-developers This is currently showing the divider line rather than disabled/empty QB bar. Is that what we decided on?

File: frontend/src/metabase/app.js [hunk 1]
```diff
@@ -53,7 +54,10 @@ function init() {
     // we shouldn't redirect these URLs because we want to handle them differently
     let WHITELIST_FORBIDDEN_URLS = [
         /api\/card\/\d+\/query$/,
-        /api\/database\/\d+\/metadata$/
+        /api\/database\/\d+\/metadata$/,
+        /api\/database\/\d+\/fields/,
+        /api\/table\/\d+\/query_metadata$/,
+        /api\/table\/\d+\/fks$/
```
Comment (salsakran) [inline frontend/src/metabase/app.js]: this seems to conflate an un-authenticated user with a user who is not authorized for the specific url they are trying to access. should we be using 401 for un-authenticated users and reserve 403 for resources that are forbidden for the specific user (or anonymous user)?

Comment (tlrobinson) [inline frontend/src/metabase/app.js]: I don't understand, that's what we do.

**Review by kdoh — APPROVED** [review]

File: frontend/src/metabase/app.js (hunk 1 above)
Comment (salsakran) [inline frontend/src/metabase/app.js]: nvm.

Comment (camsaul) [inline frontend/src/metabase/app.js]: 👍

**Review by mazameli — COMMENTED** [review]
Works as advertised from a UX standpoint. Nice job.

**Pull request merged**